.pre-commit-config.yaml

```diff
@@ -40,9 +40,10 @@ repos:
       - id: cmake-format
         name: CMake Format
         language: python
-        entry: bash -c "pip install cmake-format && python run-cmake-format.py --check"
-        entry: echo
-        files: ^(.*/CMakeLists.txt|.*.cmake)$
+        entry: cmake-format --in-place --autosort=false
```

Comment:
Why do you use raw `cmake-format` instead of our `run-cmake-format.py` wrapper?
We have configuration in the wrapper.

Reply:
Hi @kou 

It does the same thing. If you do it this way, then `pre-commit` will figure out which files to run it on, and it'll be much, much faster:
```
$ time pre-commit run cmake-format --all-files
[WARNING] Unexpected key(s) present on git://github.com/pre-commit/pre-commit-hooks: sha
CMake Format.............................................................Passed

real    0m4.472s
user    0m18.011s
sys     0m0.259s


$ time python run-cmake-format.py
WARNING config_util.py:307: The following configuration options were ignored:
  max_subargs_per_line
real    0m10.620s
user    0m10.594s
sys     0m0.017s
```

As far as I can tell, then only configuration in the wrapper is `--in-place --autosort=false` - have I missed anything else?

Reply:
Also, like this, if you enable pre-commit (`pre-commit install`) then this'll run only on files you've staged, rather than on the wholecodebase each time (as it would if using `run-cmake-format.py`). You can still choose to run it on the whole codebase though, with `--all-files` (as in the example i posted above)

Reply:
I couldn't reproduce the difference on my environment:

```console
$ time pre-commit run cmake-format --all-files
[WARNING] Unexpected key(s) present on git://github.com/pre-commit/pre-commit-hooks: sha
CMake Format.............................................................Passed

real	0m3.476s
user	0m13.000s
sys	0m0.297s
$ time python3 run-cmake-format.py

real	0m1.358s
user	0m1.308s
sys	0m0.046s
```

Reply:
:thinking: not sure why that would be, but point taken. Though there would still be a difference if you run it as a git commit hook (i.e. if you've done `pre-commit install` in your environment) because like this it'll only run on staged files, while the Python script runs on all of them each time, so this'll give faster feedback to devs

Reply:
It seems that `pre-commit` runs `cmake-format` for each target file.
But `run-cmake-format.py` runs one `cmake-format` for all target files.

Reply:
We don't want to reformat imported CMake files: https://github.com/apache/arrow/blob/master/run-cmake-format.py#L28-L29
( @xhochy Do you still want to avoid reformatting imported CMake files? The comment was added by you at 79c93c77542524042e26761c5107127f51e5c193. )
Could you use `run-cmake-format.py` to keep the configuration for it in one place?

BTW, we can't maintain the target list by current allow list style. The `patterns` is out of date. It still has renamed files. Can we change the target list to deny list style? Here is a list of imported CMake files:

* `cpp/cmake_modules/FindNumPy.cmake`
* `cpp/cmake_modules/FindPythonLibsNew.cmake`
* `cpp/cmake_modules/UseCython.cmake` (Oh... This is already reformatted...)

If `run-cmake-format.py` accepts a target file like `run-cmake-format.py cpp/CMakeLists.txt` and it processes only the given file, can we remove the performance difference between `run-cmake-format.py` and `cmake-format`?

Reply:
sure, will get back to this by the end of the will and will open a jira ticket as instructed

Reply:
you could use `pass_filenames: false` in pre-commit so that it'll just run `python run-cmake-format.py`, like this configuration will all be there

> It seems that pre-commit runs cmake-format for each target file.

I don't think so, it should just run `cmake-format file1 file2 ... filen`

Reply:
@xhochy Do you still want to avoid reformatting imported CMake files? 

Yes, this helps with diffing against their upstream. We did do changes to them for our use case and this makes updating them simpler.